File: sample_project/php/Node.php
<?php
namespace AStar;

class Node implements Comparable {
    public ?Node $parent;
    public int $x;
    public int $y;
    public float $g;
    public float $h;

    public function __construct(?Node $parent, int $x, int $y, float $g, float $h) {
        $this->parent = $parent;
        $this->x = $x;
        $this->y = $y;
        $this->g = $g;
        $this->h = $h;
    }

    public function compareTo($other): int {
        $thisF = $this->g + $this->h;
        $otherF = $other->g + $other->h;
        return $thisF <=> $otherF;
    }
}

File: sample_project/csharp/Node.cs
namespace AStarPathfinding
{
    public class Node : IComparable<Node>
    {
        public Node? Parent { get; set; }
        public int X { get; set; }
        public int Y { get; set; }
        public double G { get; set; }
        public double H { get; set; }

        public Node(Node? parent, int x, int y, double g, double h)
        {
            Parent = parent;
            X = x;
            Y = y;
            G = g;
            H = h;
        }

        public int CompareTo(Node? other)
        {
            if (other == null) return 1;
            return (G + H).CompareTo(other.G + other.H);
        }
    }
}
